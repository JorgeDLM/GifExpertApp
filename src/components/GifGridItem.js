import React from 'react'
// para importar propTypes rapido: imp tab
import PropTypes from 'prop-types'

export const GifGridItem = ({title, url, id, ultimaCategoria, categorias}) => {
    return (
        <div className={"card"}>
            <img style={{}} src={url} alt={title}/>
            <p style={{fontWeight: 'bold'}}>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

GifGridItem.defaultValue = {
    title: 'Hola'
}


export default GifGridItem