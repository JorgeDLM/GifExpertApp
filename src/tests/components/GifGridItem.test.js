import React from 'react'
import GifGridItem from "../../components/GifGridItem"
// asegurarme que se importe de enzyme
import { shallow } from "enzyme"

describe('GifGridItem', () => {
    const title = 'Rick and Morty'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem url={ url } title={ title }  />)

    test('debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )
    })

    test('debe tener la imagen con url y title', () => {
        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe( url )
    })
    
    
})
