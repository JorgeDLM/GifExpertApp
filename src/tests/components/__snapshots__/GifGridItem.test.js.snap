// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Rick and Morty"
    src="https://localhost/algo.jpg"
    style={Object {}}
  />
  <p
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Rick and Morty
  </p>
</div>
`;
